### PR TITLE
First draft spatial Decomposition updates

### LINKED ISSUE #230: Update Spatial Decomposition Templates
## Description & Tasks
New generic spatial structure means the templates remain correct but the documentation needs updating to provide the minimal/best practice structures

- [x] Update Documentation

### Discussion Points

1.  what should be the minimum spatial structure?
2. Facilities should be able to be composed in facility parts unless the it is the first child of the Context?

### Links
[Miro Element](https://miro.com/app/board/o9J_l5rJCLk=/?moveToWidget=3074457362316376620&cot=14)
[Miro Task Card](https://miro.com/app/board/o9J_l5rJCLk=/?moveToWidget=3074457362911087563&cot=14)

### Decisions 
TBC

### PR DESCRIPTION
**Pull request by AlexBrad1eyCT**
Fixes #230 

First Draft of documentation changes to address new form of spatial decomposition

### CODE REVIEW

File: IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
 
-In addition a more general hierarchical tree of spatial elements can be created by using _IfcSpatialZone_, from high to low: _IfcProject_, _IfcSite_, and _IfcSpatialZone_ with _IfcSite_ being an optional level.
+Where possible the relevent subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with preefined or user defined type is not available the higher level generic _IfcFacility_ entity can be instantiated with the relevent and agreed typing identifier defined in _IfcFacility.ObjectType_. this allows for nearly full coverage of built environment domains and/or scenarios that have yet to be addressed with specific extensions.
```
Comment (larswik) [inline IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md:7]:
```suggestion
Where possible the relevent subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with preefined or user defined type is not available the higher level generic _IfcFacility_ entity can be instantiated with the relevant and agreed typing identifier defined in _IfcFacility.ObjectType_. This allows for nearly full coverage of built environment domains and/or scenarios that have yet to be addressed with specific extensions.
```

**Review by larswik — CHANGES_REQUESTED** [review]

File: IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
+Where possible the relevent subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with preefined or user defined type is not available the higher level generic _IfcFacility_ entity can be instantiated with the relevant and agreed typing identifier defined in _IfcFacility.ObjectType_. This allows for nearly full coverage of built environment domains and/or scenarios that have yet to be addressed with specific extensions.
 
-> NOTE&nbsp; The more general hiearchical tree has been introduced as an intermediate solution and stub for further extensions to support infrastructure works.
+In addition to the identified spatial structure elements _IfcSpatialZone_ can be used to provide cross domain or functional zones within a project, these elements can be mounted into the hierarchy in the same manner as other spatial structure elements, but should not aggregate any other spatial structure element types except other _IfcSpatialZone_ instances.
```
Comment (MatthieuPERIN) [inline IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md:9]: Not agree, as far as I know the relation to mount SpatialZone into SpatialStructure is with IfcRelContainedInSpatialStructure and not IfcRelAggragate, that should be stated clearly ?

**Review by MatthieuPERIN — COMMENTED** [review]
Need to clarify the part about IfcSpatialZone and "mounting" it inside Spatial structure.

  
I can provide a remplacement text if needed !

File: IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md (hunk 2 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md]: You are correct @MatthieuPERIN, my mistake i miss read our diagram and have updated this.

File: IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
 
-In addition a more general hierarchical tree of spatial elements can be created by using _IfcSpatialZone_, from high to low: _IfcProject_, _IfcSite_, and _IfcSpatialZone_ with _IfcSite_ being an optional level.
+Where possible the relevent subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with predefined or user defined type is not available the higher level generic _IfcFacility_ entity can be instantiated with the relevant and agreed typing identifier defined in _IfcFacility.ObjectType_. This allows for nearly full coverage of built environment domains and/or scenarios that have yet to be addressed with specific extensions.
```
Comment (larswik) [inline IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md:7]:
```suggestion
Where possible the relevant subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with predefined or user defined type is not available the higher level, generic _IfcFacility_ entity can be instantiated with the relevant and agreed typing identifier defined in _IfcFacility.ObjectType_. This allows for nearly full coverage of built environment domains and/or scenarios that have yet to be addressed with specific extensions.
```

File: IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md [hunk 4]
```diff
@@ -1,9 +1,9 @@
...
 
-> NOTE&nbsp; The more general hiearchical tree has been introduced as an intermediate solution and stub for further extensions to support infrastructure works.
+In addition to the identified spatial structure elements _IfcSpatialZone_ can be used to provide cross domain or functional zones within a project, these elements are mounted into the hierarchy using the _Spatial Containment_ concept (_IfcRelContainedInSpatialStructure_) and can be aggregated into a functional hierarchy in the same manner as other spatial structure elements with the constraint that an _IfcSpatialZone_ can only composed under another _IfcSpatialZone_.
```
Comment (larswik) [inline IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md:9]: "mounted" => "included"?

File: IFC4x3/Templates/Object Composition/Aggregation/Spatial Decomposition/Documentation.md [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
 
-In addition a more general hierarchical tree of spatial elements can be created by using _IfcSpatialZone_, from low to high: _IfcSpatialZone_, _IfcSite_, and _IfcProject_, with _IfcSite_ being an optional level.
+Where possible the relevent subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with predefined or user defined type is not available the higher level generic _IfcFacility_ entity can be instantiated with the relevant and agreed typing identifier defined in _IfcFacility.ObjectType_. This allows for nearly full coverage of built environment domains and/or scenarios that have yet to be addressed with specific extensions.
```
Comment (larswik) [inline IFC4x3/Templates/Object Composition/Aggregation/Spatial Decomposition/Documentation.md:7]:
```suggestion
Where possible the relevant subtype of _IfcFacility_ should be used to describe the spatial structure element in question. When a adequate subtype of _IfcFacility_ with predefined or user defined type is not available the higher level, generic _IfcFacility_ entity can be instantiated with the relevant and agreed typing identifier defined in _IfcFacility.ObjectType_. This allows for nearly full coverage of built environment domains and/or scenarios that have yet to be addressed with specific extensions.
```

File: IFC4x3/Templates/Object Composition/Aggregation/Spatial Decomposition/Documentation.md [hunk 6]
```diff
@@ -1,9 +1,9 @@
...
 
-> NOTE&nbsp; The more general hiearchical tree has been introduced as an intermediate solution and stub for further extensions to support infrastructure works.
+In addition to the identified spatial structure elements _IfcSpatialZone_ can be used to provide cross domain or functional zones within a project, these elements are mounted into the hierarchy using the _Spatial Containment_ concept (_IfcRelContainedInSpatialStructure_) and can be aggregated into a functional hierarchy in the same manner as other spatial structure elements with the constraint that an _IfcSpatialZone_ can only composed under another _IfcSpatialZone_.
```
Comment (larswik) [inline IFC4x3/Templates/Object Composition/Aggregation/Spatial Decomposition/Documentation.md:9]: "mounted" => "included"??

**Review by larswik — CHANGES_REQUESTED** [review]

File: IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md (hunk 4 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Templates/Object Composition/Aggregation/Spatial Composition/Documentation.md:9]:
```suggestion
In addition to the identified spatial structure elements _IfcSpatialZone_ can be used to provide cross domain or functional zones within a project, these elements are included into the hierarchy using the _Spatial Containment_ concept (_IfcRelContainedInSpatialStructure_) and can be aggregated into a functional hierarchy in the same manner as other spatial structure elements with the constraint that an _IfcSpatialZone_ can only composed under another _IfcSpatialZone_.
```

File: IFC4x3/Templates/Object Composition/Aggregation/Spatial Decomposition/Documentation.md (hunk 6 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Templates/Object Composition/Aggregation/Spatial Decomposition/Documentation.md:9]:
```suggestion
In addition to the identified spatial structure elements _IfcSpatialZone_ can be used to provide cross domain or functional zones within a project, these elements are included into the hierarchy using the _Spatial Containment_ concept (_IfcRelContainedInSpatialStructure_) and can be aggregated into a functional hierarchy in the same manner as other spatial structure elements with the constraint that an _IfcSpatialZone_ can only composed under another _IfcSpatialZone_.
```

**Comment by AlexBrad1eyCT** [comment]
Changes have been submitted a review reminder for @MatthieuPERIN @SergejMuhic @larswik so we can merge this CT In

**Review by larswik — APPROVED** [review]

File: IFC4x3/Templates/Object Composition/Aggregation/Spatial Decomposition/Documentation.md [hunk 7]
```diff
@@ -1,9 +1,9 @@
...
 
-The order of spatial structure elements being included in the concept for builing projects are from low to high level: _IfcSpace_, _IfcBuildingStorey_, _IfcBuilding_, _IfcSite_ and _IfcProject_ with _IfcSite_, _IfcBuildingStorey_ and _IfcSpace_ being optional levels. Therefore an spatial structure element can only be part of an element at the same or higher level.
+The project spatial structure may be made up of a selection of different spatial structure elements with the most generic and simplest form from low to high level as follows: _IfcSpace_, _IfcFacilityPart_ (or _IfcBuildingStorey in the case of buildings), _IfcFacility_ (or any of its subtypes), _IfcSite_ and  _IfcProject_ with _IfcSite_, _FacilityPart_ and _IfcSpace_ being optional levels. Therefore a spatial structure element should only be part of an element at the same or higher level, with the exception of _IfcFacility_ which can be part of an _IfcFacilityPart_ to allow for the regional or longitudinal division of a higjh level facility in to sections which can then contain one or more smaller functional facilities.
```
Comment (MatthieuPERIN) [inline IFC4x3/Templates/Object Composition/Aggregation/Spatial Decomposition/Documentation.md:5]: Typos:
higjh -> high

**Review by MatthieuPERIN — CHANGES_REQUESTED** [review]
One typo, the rest is perfect !

**Review by MatthieuPERIN — APPROVED** [review]
Perfect !

**Review by larswik — APPROVED** [review]

**Pull request merged**